x-pack/solutions/search/plugins/search_query_rules/public/components/query_ruleset_detail/query_rule_detail_panel.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiButton, EuiText } from '@elastic/eui';
+import { KibanaPageTemplate } from '@kbn/shared-ux-page-kibana-template';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { QueryRulesQueryRule } from '@elastic/elasticsearch/lib/api/types';
+import { QueryRuleDraggableList } from './query_rule_draggable_list';
+
+interface QueryRuleDetailPanelProps {
+  rules: QueryRulesQueryRule[];
+  setRules: (rules: QueryRulesQueryRule[]) => void;
+}
+export const QueryRuleDetailPanel: React.FC<QueryRuleDetailPanelProps> = ({ rules, setRules }) => {
+  return (
+    <KibanaPageTemplate.Section restrictWidth>
+      <EuiFlexGroup justifyContent="spaceBetween" direction="column">
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+            <EuiFlexItem grow={false}>
+              <EuiFlexGroup alignItems="center">
+                <EuiFlexItem>
+                  <EuiButton
+                    iconType="plusInCircle"
+                    color="primary"
+                    data-test-subj="queryRulesetDetailAddRuleButton"
+                    onClick={() => {
+                      // Logic to add a new rule
+                    }}
+                  >
+                    <FormattedMessage
+                      id="xpack.queryRules.queryRulesetDetail.addRuleButton"
+                      defaultMessage="Add rule"
+                    />
+                  </EuiButton>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiText>
+                    {rules.length > 1 ? (
```

Comment:
nitpick: you can do this inline with `FormattedMessage` [message syntax](https://formatjs.github.io/docs/core-concepts/icu-syntax#plural-format)

```
<FormattedMessage
  id="xpack.queryRules.queryRulesetDetail.ruleCount"
  defaultMessage="{ruleCount, plural, one {# rule} other {# rules}}"
  values={{ ruleCount: rules.length }}
  data-test-subj="queryRulesetDetailRuleCount"
/>
```